lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1903,7 +1903,7 @@ sub _job_stop_cluster {
     return 0 if $job == $self->id;
     my $rset = $self->result_source->resultset;
 
-    $job = $rset->search({id => $job, result => NONE})->first;
+    $job = $rset->find({id => $job, result => NONE});
```

Comment:
I tried this in my PR, but it doesn't work. `find` wants primary/unique keys. any additional search fields will be ignored.

Reply:
but @Martchus in https://github.com/os-autoinst/openQA/pull/4202#discussion_r828210781 said we could use find with the `rows => 1` attribute. So what is correct now?

Reply:
Well, try it out with `DBIC_TRACE=1` and see `WHERE ( me.id = ? )` without mentioning `result`. That's obviously wrong, isn't it?

This is about the jobs table, which has a primary key `id`. *Additional* columns in `find()` will be ignored. (and yes, I think that should die or at least warn instead of silently ignore it.)

https://github.com/os-autoinst/openQA/pull/4202#discussion_r828210781is about the `job_modules` table, where `name` is *part* of a unique key constraint.
So these are two different situations.

Tests are failing, and I haven't had a closer look, but this line here could be the problem.